tests/eclient/testdata/app_dhcp.txt

```diff
@@ -0,0 +1,169 @@
+! stderr .
+
+# We had a bug that DHCPACK returned for DHCPINFORM would result
+# in EVE replacing the recorded IP address with unspecified IP 0.0.0.0.
+# This is because RFC2131 states (in section 4.3.5) that yiaddr inside
+# DHCPACK should be empty in this case and EVE uses this field to learn
+# IP assignments.
+exec -t 2m bash send_dhcpinform.sh $eclient_mac $eclient_ip
+exec sleep 60
+exec -t 2m bash wait_for_ip_assignment.sh $eclient_mac $eclient_ip
+! stderr .
+
+message 'Resource cleaning'
+eden pod delete eclient
+eden pod delete dhcp-server
+test eden.app.test -test.v -timewait 10m - eclient dhcp-server
+
+eden network delete switch
+eden network delete nat
+test eden.network.test -test.v -timewait 10m - switch nat
+stdout 'no network with switch found'
+stdout 'no network with nat found'
+eden network ls
+! stdout '^switch\s'
+! stdout '^nat\s'
+
+-- check_vm_support.sh --
+EDEN={{EdenConfig "eden.root"}}/{{EdenConfig "eden.bin-dist"}}/{{EdenConfig "eden.eden-bin"}}
+
+:>.env
+while true;
+do
+    virt=$($EDEN info --out InfoContent.dinfo.Capabilities.HWAssistedVirtualization | tail -n 1)
+    if [ -z "$virt" ]; then
+        sleep 3
+        continue
+    fi
+    [ "$virt" == "true" ] && echo "with_hw_virt=true" >>.env
+    break
+done
+
+-- wait_ssh.sh --
+EDEN={{EdenConfig "eden.root"}}/{{EdenConfig "eden.bin-dist"}}/{{EdenConfig "eden.eden-bin"}}
+HOST=$($EDEN eve ip)
+
+for i in `seq 20`
+do
+  sleep 20
+  # Test SSH-access to container
+  echo {{template "ssh"}}$HOST grep -q Ubuntu /etc/issue
+  {{template "ssh"}}$HOST grep -q Ubuntu /etc/issue && break
+done
+
+-- wait_for_app_ip.sh --
+#!/bin/sh
+
+APP=$1
+EDEN={{EdenConfig "eden.root"}}/{{EdenConfig "eden.bin-dist"}}/{{EdenConfig "eden.eden-bin"}}
+HOST=$($EDEN eve ip)
+
+function get_ip {
+    ifconfig=$({{template "ssh"}}$HOST ifconfig 2>/dev/null) || return 1
+    if [ -z "$ifconfig" ]; then
+        return 1
+    fi
+    interfaces=$(echo "$ifconfig" | grep "^\w" | grep -v LOOPBACK | cut -d: -f1)
+    eth1=$(echo "$interfaces" | awk 'NR==2')
+    IP=$(echo "$ifconfig" | grep "^${eth1}: " -A 1 | awk '/inet / {print $2}')
+    MAC=$(echo "$ifconfig" | grep "^${eth1}: " -A 2 | awk '/ether / {print $2}')
+    if [ ! -z "$IP" ]; then
+        echo "IP address of $APP is: $IP"
+        echo "${APP}_ip=$IP" >.env
+        echo "MAC address of $APP is: $MAC"
+        echo "${APP}_mac=$MAC" >>.env
+        return 0
+    fi
+    return 1
+}
+
+until get_ip; do sleep 3; done
+
+-- wait_for_ip_assignment.sh --
+#!/bin/sh
+
+EXP_MAC=$1
+EXP_IP=$2
+EDEN={{EdenConfig "eden.root"}}/{{EdenConfig "eden.bin-dist"}}/{{EdenConfig "eden.eden-bin"}}
+
+function get_ip_assignments {
+    assignments=$($EDEN info InfoContent.niinfo.displayname:switch --tail 1 --out InfoContent.niinfo.ipAssignments)
+    echo "Current IP assignments: $assignments"
+    if echo "$assignments" | grep -q "macAddress:\"$EXP_MAC\"  ipAddress:\"$EXP_IP\""; then
```

Comment:
It seems that double space is not the stable output, can you please use ` \+` between mac and ip here?

Reply:
done